make/default_compiler_options

```diff
@@ -0,0 +1,19 @@
+# Set default compiler options.
+# Users should only need to set these three variables for use.
+# - CC: The compiler to use. Expecting g++ or clang++.
+# - O: Optimization level. Valid values are {0, 1, 2, 3}.
+# - AR: archiver (must specify for cross-compiling)
+##
+CC = clang++
+O = 3
+O_STANC = 0
+AR = ar
+
+# CPPFLAGS are used for both C and C++ compilation
```

Comment:
I know, but it seems to be used in this way sometimes. We couldn't duplicate CXXFLAGS to CFLAGS; the original problem is that non-C++ compilers reject CXXFLAGS like `-std=c++1y`. The alternative might be having some fourth variable that we internally use as CPPFLAGS and then copying it at the end of every makefile to CXXFLAGS and CFLAGS.
I think we should just use CPPFLAGS; I've seen this usage before in the wild. See e.g. https://stackoverflow.com/questions/2754966/cflags-vs-cppflags